rc2/src/lib.rs

```diff
@@ -0,0 +1,199 @@
+        *j -= 1;
+
+        r[2] = (r[2] << 13) | (r[2] >> 3);
+        r[2] = r[2].wrapping_sub(self.exp_key[*j]).wrapping_sub(r[1] & r[0]).wrapping_sub(!r[1] & r[3]);
+        *j -= 1;
+
+        r[1] = (r[1] << 14) | (r[1] >> 2);
+        r[1] = r[1].wrapping_sub(self.exp_key[*j]).wrapping_sub(r[0] & r[3]).wrapping_sub(!r[0] & r[2]);
+        *j -= 1;
+
+        r[0] = (r[0] << 15) | (r[0] >> 1);
+        r[0] = r[0].wrapping_sub(self.exp_key[*j]).wrapping_sub(r[3] & r[2]).wrapping_sub(!r[3] & r[1]);
+        *j = j.wrapping_sub(1);
+    }
+
+    fn reverse_mash(&self, r: &mut [u16; 4]) {
+        r[3] = r[3].wrapping_sub(self.exp_key[(r[2] & 63) as usize]);
+        r[2] = r[2].wrapping_sub(self.exp_key[(r[1] & 63) as usize]);
+        r[1] = r[1].wrapping_sub(self.exp_key[(r[0] & 63) as usize]);
+        r[0] = r[0].wrapping_sub(self.exp_key[(r[3] & 63) as usize]);
+    }
+
+    fn encrypt(&self, block: [u8; 8]) -> [u8; 8] {
+        let mut r: [u16; 4] = [0; 4];
+
+        r[0] = ((block[1] as u16) << 8) + (block[0] as u16);
+        r[1] = ((block[3] as u16) << 8) + (block[2] as u16);
+        r[2] = ((block[5] as u16) << 8) + (block[4] as u16);
+        r[3] = ((block[7] as u16) << 8) + (block[6] as u16);
+
+        let mut j = 0;
+
+        for i in 0..16 {
+            self.mix(&mut r, &mut j);
+            if i == 4 || i == 10 {
+                self.mash(&mut r);
+            }
+        }
+
+        let mut ct: [u8; 8] = [0; 8];
+        ct[0] = r[0] as u8;
+        ct[1] = (r[0] >> 8) as u8;
+        ct[2] = r[1] as u8;
+        ct[3] = (r[1] >> 8) as u8;
+        ct[4] = r[2] as u8;
+        ct[5] = (r[2] >> 8) as u8;
+        ct[6] = r[3] as u8;
+        ct[7] = (r[3] >> 8) as u8;
+
+        ct
+    }
+
+    fn decrypt(&self, block: [u8; 8]) -> [u8; 8] {
+        let mut r: [u16; 4] = [0; 4];
+
+        r[0] = ((block[1] as u16) << 8) + (block[0] as u16);
+        r[1] = ((block[3] as u16) << 8) + (block[2] as u16);
+        r[2] = ((block[5] as u16) << 8) + (block[4] as u16);
+        r[3] = ((block[7] as u16) << 8) + (block[6] as u16);
+
+        let mut j = 63;
+
+        for i in 0..16 {
+            self.reverse_mix(&mut r, &mut j);
+            if i == 4 || i == 10 {
+                self.reverse_mash(&mut r);
+            }
+        }
+
+        let mut ct: [u8; 8] = [0; 8];
+        ct[0] = r[0] as u8;
+        ct[1] = (r[0] >> 8) as u8;
+        ct[2] = r[1] as u8;
+        ct[3] = (r[1] >> 8) as u8;
+        ct[4] = r[2] as u8;
+        ct[5] = (r[2] >> 8) as u8;
+        ct[6] = r[3] as u8;
+        ct[7] = (r[3] >> 8) as u8;
+
+        ct
+    }
+}
+
+impl BlockCipherVarKey for RC2 {
+    fn new(key: &[u8]) -> RC2 {
+        RC2 {
+            exp_key: RC2::expand_key(key, key.len()*8)
+        }
+    }
+}
+
+impl BlockCipher for RC2 {
+    type BlockSize = U8;
+
+    fn encrypt_block(&self, input: &Block<U8>, output: &mut Block<U8>) {
+        let mut iblock: [u8; 8] = [0; 8];
+        for i in 0..8 {
```

Comment:
This method can be shortened to just one line `self.encrypt(input)` if you'll add changes described in the comment to `encrypt` method. Same goes for `decrypt_block` and `decrypt`.